### PR TITLE
Add structural validation for recommendation rules

### PR DESCRIPTION
**Pull request by THEBLESSMAN867**
## Summary
- enforce structural validation of recommendation rules to ensure thresholds, discriminants, and templates include rigorous, non-rhetorical details
- add reusable strict template fixtures to the recommendation coverage tests and update cases to satisfy the stronger validation logic

## Testing
- pytest tests/test_recommendation_coverage.py *(fails: pyenv reports Python 3.11.9 not installed in this environment)*

https://chatgpt.com/codex/tasks/task_e_6906dcf848cc8328920db9300e0cd5db

## Summary by CodeRabbit

* **New Features**
  * Added a validation layer for recommendation rules to improve data integrity and loading behavior
  * Recommendation API now accepts explicit context and richer optional fields for more detailed outputs

* **Bug Fixes**
  * Ensures only structurally valid rules are registered, reducing invalid recommendations at runtime

* **Tests**
  * Expanded and hardened tests with a reusable strict template to improve coverage and isolation

___

## **CodeAnt-AI Description**
**Enforce strict structural validation for recommendation rules**

### What Changed
- Rule files are validated when loaded and loading fails with explicit validation errors if rules are missing required fields, have non-numeric or out-of-range thresholds, or use invalid discriminants.
- Recommendation templates must include actionable "problem" and "intervention" text, a measurable indicator (name, numeric target, unit), responsible parties including partners, a horizon (start/end), and concrete verification artifacts; templates that lack these are rejected.
- Level-specific checks prevent invalid rules: MICRO rules require pa/dim identifiers and numeric score thresholds within 0–3; MESO rules require a cluster identifier plus at least one valid discriminant (score band, variance, or weak PA); MACRO rules must declare at least one macro discriminant (clusters below target, macro band, variance alert, or priority gaps).
- Tests updated to provide a reusable strict template fixture, to exercise variable substitution, large-rule batches, boundary conditions, and to replace previously minimal templates with the stricter template so only well-formed rules pass.

### Impact
 `✅ Clearer rule validation errors during rule file load`
 `✅ Fewer non-actionable recommendations accepted`
 `✅ Fewer runtime failures from malformed thresholds or missing discriminants`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: src/saaaaaa/analysis/recommendation_engine.py [hunk 1]
```diff
@@ -565,6 +565,216 @@ def _substitute_variables(self, text: str, substitutions: Dict[str, str]) -> str
...
+        pa_id = when['pa_id']
+        dim_id = when['dim_id']
+        if not isinstance(pa_id, str) or not pa_id.strip():
+            raise ValueError(f"Rule {rule_id} has invalid pa_id")
+        if not isinstance(dim_id, str) or not dim_id.strip():
+            raise ValueError(f"Rule {rule_id} has invalid dim_id")
+
+        score_lt = when['score_lt']
+        if not self._is_number(score_lt):
+            raise ValueError(f"Rule {rule_id} has non numeric MICRO threshold")
```
[the remark this answers is not in the document]
Comment (tesislizayjuan-debug) [reply src/saaaaaa/analysis/recommendation_engine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kkkkknhh/SAAAAAA/pull/195#discussion_r2507158558)
